specs/chugsplash-manager.md

```diff
@@ -0,0 +1,178 @@
+* The ChugSplashManager MUST increase the `totalDebt` by the `BUNDLE_EXECUTION_BOND_AMOUNT`.
+* The ChugSplashManager MUST emit the event ChugSplashBundleCancelled with the ID of the bundle and the total number of actions executed. The ChugSplashManager MUST announce this event to the registry.
+
+## Withdrawing ETH
+
+* ETH stored in the contract MUST only be withdrawable by the `admin` of the `ChugSplashManager`.
+* The admin ETH withdrawal function MUST NOT be callable while a bundle is in the `APPROVED` state.
+* The admin ETH withdrawal function MUST only allow the admin to withdraw the balance of the account minus `totalDebt`.
+* The ChugSplashManager MUST emit the ETHWithdrawn event with the recipient and the amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Claiming executor payments
+
+* Executors MUST be able to claim payments at any time.
+* Executors MUST NOT be able to claim more payments than their `debt` amount.
+* The ChugSplashManager MUST update the `debt` and `totalDebt` values to reflect the payment.
+* The ChugSplashManager MUST emit the ExecutorPayment event with the recipient and amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Depositing ETH
+
+* Any address should be able to deposit ETH into the `ChugSplashManager` contract at any time.
+
+## Claiming a bundle
+
+* Anyone should be able to claim a bundle.
+* The executor's `debt` plus the `msg.value` MUST be greater than or equal to the `executorBondAmount`.
+* The bundle being claimed MUST have an `APPROVED` status.
+* The current `block.timestamp` MUST be greater than the `timeClaimed` plus the `executionLockTime`.
+* Claiming a bundle MUST set the `timeClaimed` to be the `block.timestamp`.
+* Claiming a bundle MUST set the `selectedExecutor` to be the executor that claimed the bundle.
+* The ChugSplashManager MUST increase the `debt` owed to the current executor by the `executorBondAmount`.
+* The ChugSplashManager MUST increase the `totalDebt` by the `executorBondAmount`.
+* The ChugSplashManager MUST emit the event ChugSplashBundleClaimed with the claimed bundle ID and the address of the executor. The ChugSplashManager MUST announce this event to the registry.
+
+## Setting the executor's bond amount
+
+* The `admin` of the `ChugSplashManager` MUST be the only address that can set the `executorBondAmount`.
+* The executor bond amount MUST NOT be changed while there is an active bundle.
+* MUST set `executorBondAmount` to its new value.
+* The ChugSplashManager MUST emit the ExecutorBondAmountSet event with the new `executorBondAmount`. The ChugSplashManager MUST announce this event to the registry.
+
+## Transferring proxy ownership from the ChugSplashManager
+
+* The `admin` of the `ChugSplashManager` MUST be the only address that can transfer proxy ownership from the ChugSplashManager.
+* The proxy MUST already be deployed.
+* The owner of the proxy MUST be the ChugSplashManager.
```

Comment:
I don't think this line and the line above are necessary to specify -- if the user tries to transfer something that the `ChugSplashManager` doesn't control then it'll just fail, which is fine.